hoomd/md/TwoStepRATTLEBD.h

```diff
@@ -0,0 +1,476 @@
+#include "QuaternionMath.h"
+#include "hoomd/HOOMDMath.h"
+#include "TwoStepRATTLENVE.h"
+
+#include "hoomd/RandomNumbers.h"
+#include "hoomd/RNGIdentifiers.h"
+using namespace hoomd;
+
+#ifndef __TWO_STEP_RATTLE_BD_H__
+#define __TWO_STEP_RATTLE_BD_H__
+
+/*! \file TwoStepRATTLEBD.h
+    \brief Declares the TwoStepRATTLEBD class
+*/
+
+#ifdef NVCC
+#error This header cannot be compiled by nvcc
+#endif
+
+#include <pybind11/pybind11.h>
+
+
+//! Integrates part of the system forward in two steps with Brownian dynamics
+/*! Implements RATTLE applied on Brownian dynamics.
+
+    Brownian dynamics modifies the Langevin equation by setting the acceleration term to 0 and assuming terminal
+    velocity.
+
+    \ingroup updaters
+*/
+template < class Manifold>
+class PYBIND11_EXPORT TwoStepRATTLEBD : public TwoStepLangevinBase
+    {
+    public:
+        //! Constructs the integration method and associates it with the system
+        TwoStepRATTLEBD(std::shared_ptr<SystemDefinition> sysdef,
+                    std::shared_ptr<ParticleGroup> group,
+                    Manifold manifold,
+                    std::shared_ptr<Variant> T,
+                    unsigned int seed,
+                    Scalar eta = 0.000001
+                    );
+
+        virtual ~TwoStepRATTLEBD()
+        {
+        m_exec_conf->msg->notice(5) << "Destroying TwoStepBD" << endl;
+        }
+
+        //! Performs the second step of the integration
+        virtual void integrateStepOne(unsigned int timestep);
+
+        //! Performs the second step of the integration
+        virtual void integrateStepTwo(unsigned int timestep);
+
+        //! Includes the RATTLE forces to the virial/net force
+        virtual void IncludeRATTLEForce(unsigned int timestep);
+
+        //! Get the number of degrees of freedom granted to a given group
+        virtual Scalar getTranslationalDOF(std::shared_ptr<ParticleGroup> group)
+        {
+        // get the size of the intersection between query_group and m_group
+        unsigned int intersect_size = ParticleGroup::groupIntersection(group, m_group)->getNumMembersGlobal();
+
+        return ( m_sysdef->getNDimensions() - 1 ) * intersect_size;
+        }
+
+        /// Sets eta
+        void setEta(pybind11::object eta){ m_eta = pybind11::cast<Scalar>(eta); };
+
+        /// Gets alpha
+        pybind11::object getEta(){ return pybind11::cast(m_eta); };
+
+    protected:
+        Manifold m_manifold;  //!< The manifold used for the RATTLE constraint
+        bool m_noiseless_t;
+        bool m_noiseless_r;
+        Scalar m_eta;                      //!< The eta value of the RATTLE algorithm, setting the tolerance to the manifold
+
+        GPUArray<Scalar3> m_f_brownian; //! Brownian random force
+    };
+
+
+
+/*! \file TwoStepRATTLEBD.h
+    \brief Contains code for the TwoStepRATTLEBD class
+    \Warning NDOF is still 3*(N_part-1) and not 2*(N_part-1)!!! Has to be considered in thermodynamic quantities calculations.
```

Comment:
Is this comment accurate? It seems that you correctly compute the number of dimensions in the overridden `getTranslationalDOF`.